src/coreclr/jit/fgbasic.cpp

```diff
@@ -4828,13 +4828,16 @@ BasicBlock* Compiler::fgSplitBlockAfterStatement(BasicBlock* curr, Statement* st
         assert(newBlock->bbCodeOffs == BAD_IL_OFFSET);
         assert(newBlock->bbCodeOffsEnd == BAD_IL_OFFSET);
 
-        // curr->bbCodeOffs remains the same
-        newBlock->bbCodeOffsEnd = curr->bbCodeOffsEnd;
+        if (!newBlock->HasFlag(BBF_INTERNAL))
+        {
+            // curr->bbCodeOffs remains the same
+            newBlock->bbCodeOffsEnd = curr->bbCodeOffsEnd;
 
-        IL_OFFSET splitPointILOffset = fgFindBlockILOffset(newBlock);
+            IL_OFFSET splitPointILOffset = fgFindBlockILOffset(newBlock);
```

Comment:
Instead of doing the `if (!newBlock->HasFlag(BBF_INTERNAL))` check, I was debating whether or not `fgFindBlockILOffset` should always return `BAD_IL_OFFSET`.